crates/sui-types/src/committee.rs

```diff
@@ -44,6 +46,15 @@ impl Committee {
         self.epoch
     }
 
+    pub fn public_key(&self, authority: &AuthorityName) -> SuiResult<PublicKey> {
+        // do we know, or can we build a valid public key?
```

Comment:
We do yes, because we check this upon construction.

Reply:
Here the `public_key` function takes an `authority` as argument, and it can be any value. So we don't know if it's a valid public key.

Reply:
Oh I see, so the issue is when you don't find the key on the cache, and add a new entry (which might not have a valid key). Who is invoking this api? Do we already ensure during the invocation flow that the `AuthorityName` is part of the committee already?

Reply:
Btw a lot of these countermeasures would be safer if we had a wrapper object called Authority that carries both the pubKey and name (key bytes). Which upon construction we always ensure it was correct.

Reply:
signedTx shouldn't in theory carry validator pubKeys but a pointer (index) to which validator in this epoch. We can address this in an upcoming PR after some co-brainstorming.